examples/2d/kymatio_parametric_scattering.py

```diff
@@ -0,0 +1,435 @@
+            self.params_filters[i] = nn.Parameter(self.params_filters[i])
+            self.register_parameter(name='scattering_params_'+str(i), param=self.params_filters[i])
+        self.params_filters = nn.ParameterList(self.params_filters)
+        self.register_buffer(name='grid', tensor=grid)
+        
+        def updateFilters_hook(self, ip):
+            """Update the filters to reflect 
+            the new parameter values obtained from gradient descent"""
+            if (self.training or self.scatteringTrain):
+                _, psi = self.load_filters()
+                wavelets = morlets(self.grid, 
+                            self.scattering_params_0,
+                            self.scattering_params_1,
+                            self.scattering_params_2, 
+                            self.scattering_params_3)
+
+                self.psi = update_psi(self.J, psi, wavelets)
+                self.register_filters()
+                self.scatteringTrain = self.training
+        self.pre_hook = self.register_forward_pre_hook(updateFilters_hook)
+
+        def reshape_hook(self, x, S):
+            S = S[:,:, -self.n_coefficients:,:,:]
+            S = S.reshape(S.size(0), -1, S.size(3), S.size(4))
+            return S
+        self.register_forward_hook(reshape_hook)
+
+
+###############################################################################
+# We visualize our wavelets prior to training and after. We look at 
+# the Littlewood Paley diagram. Dekha means "Look! See!" in Punjabi. 
+
+def littlewood_paley_dekha(S, display=True):
+    wavelets = morlets(S.grid, S.params_filters[0], 
+                                      S.params_filters[1], S.params_filters[2], 
+                                        S.params_filters[3]).detach().numpy()
+    gabors = morlets(S.grid, S.params_filters[0], 
+                                      S.params_filters[1], S.params_filters[2], 
+                                      S.params_filters[3], return_gabors=True).cpu().detach().numpy() 
+    grid = S.grid
+    lp = (np.abs(wavelets) ** 2).sum(0)
+    fig, ax = plt.subplots()
+    
+    plt.imshow(np.fft.fftshift(lp))
+    if display:
+        grid = grid + 18
+        for i in np.abs(gabors) ** 2:
+            wave = np.fft.fftshift(i)
+            ax.contour(grid[1], grid[0], wave, 1, colors='white')
+    plt.show()
+
+
+###############################################################################
+# Lets initailize our newly implemented parametric scattering and look at the 
+# Littlewood-Paley diagram.
+LS = ParametricScattering2D(J=2, M=28, N=28, L=8)
+littlewood_paley_dekha(LS)
+
+
+###############################################################################
+# Training the Parametric Scattering
+# ---------------------------------------------------------------------------------------------------------------------------------
+# We train the parametric scattering on MNSIT digits and then evaluate. 
```

Comment:
"and then evaluate and visualize the resulting learned filters"

Reply:
Perhaps we could add a comment for the littlewood_paley_dekha function, explaining that this plots the entire filterbank in Fourier. I think this would be clearer than 'Littlewood-Paley diagram'

Reply:
do you wanna do this change or shall I?